### PR TITLE
fix: run `onMount` and `onUpdate` code in browser only in angular

### PR DESCRIPTION
**Pull request by sidmohanty11**
## Description

Moved all our initializations (handling generated state) to the `onInit` hook which runs inside `ngOnInit` now (runs both on ssr and csr) but `onMount` and `onUpdate` hooks which are in turn generated inside `ngOnInit` and `ngOnChanges` now run on CSR only (has a check if its running on browser), this is to prevent running any onUpdate/onMount hooks on the server side incase of angular.

Jira
https://builder-io.atlassian.net/browse/ENG-6195

Make sure to follow the PR preparation steps in [CONTRIBUTING.md](../CONTRIBUTING.md#preparing-your-pr) before submitting your PR:

- [x] format the codebase: from the root, run `yarn fmt:prettier`.
- [x] update all snapshots (in core & CLI): from the root, run `yarn test:update`
- [x] add Changeset entry: from the root, run `yarn g:changeset` and follow the CLI instructions. Alternatively, use the Changeset Github Bot to create the file.

### CODE REVIEW

File: packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap [hunk 1]
```diff
@@ -177,67 +178,12 @@ export class MyBasicForShowComponentModule {}
...
-    const change = this.myService.method(\\"change\\");
-    console.log(change);
-  };
-  node_0_div = null;
-
-  constructor(public myService: MyService) {
-    const hi = this.myService.method(\\"hi\\");
-    console.log(hi);
-  }
+"unknown: Unexpected token (1:8)
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap:181]: something fishy is going on here

File: packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap [hunk 2]
```diff
@@ -2505,6 +2390,12 @@ export default class MyBasicForNoTagRefComponent {
   ngOnInit() {
     this.mergedInputs_0 = {};
 
+    this.mergedInputs_0 = {};
+
+    this.mergedInputs_0 = {};
+
+    this.mergedInputs_0 = {};
```
Comment (samijaber) [inline packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap:2397]: this duplication doesn't make sense. seems like we're looping over the wrong thing in the generator

File: packages/core/src/generators/angular/index.ts [hunk 3]
```diff
@@ -569,14 +550,10 @@ const handleBindings = (
...
       if (item.bindings[key]?.type !== 'spread' && !key.startsWith('on')) {
         json.state[newBindingName] = { code: 'null', type: 'property' };
-        json.hooks['onMount'].push({
-          code: `state.${newBindingName} = ${item.bindings[key]!.code}`,
-          onSSR: false,
-        });
-        json.hooks['onUpdate'] = json.hooks['onUpdate'] || [];
-        json.hooks['onUpdate'].push({
-          code: `state.${newBindingName} = ${item.bindings[key]!.code}`,
-        });
```
Comment (samijaber) [inline packages/core/src/generators/angular/index.ts:579]: I'm confused why this change is part of this PR.

IIUC, by moving this code from `onMount` and `onUpdate` to `onInit`, you're essentially no longer making these `property` state fields reactive right? 

This feels like a different unrelated change to me. Can you explain why it's made here?

File: packages/core/src/generators/angular/index.ts [hunk 4]
```diff
@@ -376,29 +376,10 @@ export const blockToAngular = ({
...
-        root.hooks.onUpdate &&
-        root.hooks.onUpdate.length > 0 &&
-        !root.hooks.onUpdate
-          .map((hook) => hook.code)
-          .join('')
-          .includes(inputsPropsStateName)
-      ) {
-        root.hooks.onUpdate.push({
-          code: `this.${inputsPropsStateName} = {${allProps}}`,
-        });
```
Comment (samijaber) [inline packages/core/src/generators/angular/index.ts:400]: Same question here on why this change is part of this PR

File: packages/core/src/generators/angular/index.ts (hunk 3 above)
Comment (sidmohanty11) [reply packages/core/src/generators/angular/index.ts:579]: will revert

File: packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap (hunk 1 above)
Comment (sidmohanty11) [reply packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap]: yes found out that, this is because of not adding `\n` to onInit code it was mixing up everything: something like this - `{}state.something` have fixed this by adding new lines

File: packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap (hunk 2 above)
Comment (sidmohanty11) [reply packages/core/src/__tests__/__snapshots__/angular.state.test.ts.snap]: yes! have added a check to only add this state once if it's not added before (this happens because we loop through bindings and where this state is actually used, so this depends on however many times that specific binding is being used in the code).

**Review by samijaber — APPROVED** [review]
🙏🏽
